tests/checkers/unittest_unicode/bad_chars.py

```diff
@@ -0,0 +1,255 @@
+import itertools
+from pathlib import Path
+from typing import Tuple, cast
+
+import astroid
+import pytest
+from astroid import AstroidBuildingError, nodes
+
+import pylint.checkers.unicode
+import pylint.interfaces
+import pylint.testutils
+
+from . import CODEC_AND_MSG, FakeNode
+
+
+class TestBadCharsChecker(pylint.testutils.CheckerTestCase):
+    CHECKER_CLASS = pylint.checkers.unicode.UnicodeChecker
+
+    checker: pylint.checkers.unicode.UnicodeChecker
+
+    @pytest.mark.parametrize(
+        "codec_and_msg, line_ending, add_invalid_bytes",
+        [
+            pytest.param(
+                codec_and_msg,
+                line_ending[0],
+                suffix[0],
+                id=f"{codec_and_msg[0]}_{line_ending[1]}_{suffix[1]}",
+            )
+            for codec_and_msg, line_ending, suffix in itertools.product(
+                CODEC_AND_MSG,
+                (("\n", "linux"), ("\r\n", "windows")),
+                ((False, "valid_line"), (True, "not_decode_able_line")),
+            )
+            # Only utf8 can drop invalid lines
+            if codec_and_msg[0].startswith("utf") or not suffix[0]
+        ],
+    )
+    def test_find_bad_chars_without_nul(
```

Comment:
```suggestion
    def test_find_bad_chars_without_null(
```
? I don't understand what "nul" means here.

Reply:
Historic name. 
It was to check all bad_chars but not `\nul`  because this char crashes Python/Astroid....

Reply:
OK, understood :) Maybe we could add this explanation in the test docstring so this is not lost in this MR ?

Reply:
renamed